addons/web_editor/static/src/js/editor/rte.js

```diff
@@ -698,7 +698,9 @@ var RTEWidget = Widget.extend({
              * and removed at the start and at the end of it.
              */
             let hasContentEditable = $target.attr('contenteditable');
-            $target.attr('contenteditable', true);
+            if (!$target.hasClass('o_not_editable')) {
```

Comment:
Maybe using `.closest('.o_not_editable').length` would be more accurate, a link inside an element with that class should not be editable either.